pkg/operator/ceph/object/rgw.go

```diff
@@ -331,39 +330,11 @@ func EmptyPool(pool cephv1.PoolSpec) bool {
 	return reflect.DeepEqual(pool, cephv1.PoolSpec{})
 }
 
-// GetDomainName build the dns name to reach out the service endpoint
-func GetDomainName(s *cephv1.CephObjectStore) string {
-	return getDomainName(s, true)
-}
-
 func GetStableDomainName(s *cephv1.CephObjectStore) string {
-	return getDomainName(s, false)
-}
-
-func getDomainName(s *cephv1.CephObjectStore, returnRandomDomainIfMultiple bool) string {
-	endpoints := []string{}
-	if s.Spec.IsExternal() {
```

Comment:
As part of this work, I really wanted to clean up some of the confusing function names and behavior here to focus on making sure the function name helps imply usage better.

After this, internal usage should be separated into 2 clear uses:
- get an endpoint suitable for admin ops
- get the endpoint that should be advertised to COSUsers, OBCs, COSI

The only remaining exception is the `GetStableDomainName()` which returns stable naming used for preserving legacy behavior on the status.info `endpoint`/`secureEndpoint` outputs.